File: easy/35 (Search insert position)/solution.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:

        l = len(nums)
        ans = 0
        lowest = False
        for i in range(l):
            if(nums[i] == target):
                ans = i
                return i
            if(nums[i] >= target and not lowest):
                lowest = True
                ans = i

        if(not lowest): return l
        else: return ans
            
        

File: easy/35 (Search insert position)/solution.c
int searchInsert(int* nums, int numsSize, int target) {
    bool lowest = false;
    int ans = 0;
    for(int i = 0; i < numsSize; i++){
        if(nums[i] == target){
            return i;
        }
        if(nums[i] >= target && !lowest){
            ans = i;
            lowest = true;
        }
    }
    if(!lowest) return numsSize;
    return ans;
}